docs/msource.md

```diff
@@ -0,0 +1,119 @@
+# MSource
+
+-----------
+
+## Point
+
+### Template structure
+
+### Description
+
+### Parameters
+
+### Dependencies
+
+### Products
+
+
+-----------
+
+## SeqConserved
+
+### Template structure
+
+One template argument `FImpl`, expected to be a fermion implementation.
+
+### Description
+
+Sequential source with an insertion of a conserved current $J_\mu$ using a `PropagatorField` $q(x)$ for $t_A\leq x_3 \leq t_B$:
+
+$$source(x) = \sum\limits_{\mu=\mu_{min}}^{\mu_{max}} \theta(x_3 - t_A)\theta(t_B - x_3)\exp(i x\cdot p)\,J_\mu q(x)$$
+
+Optionally an `EmField` $A_\mu$ can be inserted in addition:
+
+$$source(x) = \sum\limits_{\mu=\mu_{min}}^{\mu_{max}} \theta(x_3 - t_A)\theta(t_B - x_3)\exp(i x\cdot p)\,A_\mu \,J_\mu q(x)$$
+
+The following currents can be computed:
+
+- Wilson action
+	- Conserved Vector Current
+	$$\mathcal{V}_\mu(x) = \overline{q}(x) J_\mu q(x) = \frac{1}{2}\left[\overline{q}(x+\mu)(1+\gamma_\mu)U_\mu^\dagger(x)q(x)-\overline{q}(x)(1-\gamma_\mu)U_\mu(x)q(x+\mu)\right]$$
+
+- Domain Wall action
```

Comment:
Capital Q is undefined. In general, although I appreciate you implemented all the formulas, it can be hard to maintain. There are much more actions in Grid and some of them (e.g. Möbius) have quite gory expressions. Maybe you just want to say "examples of currents that can be inserted".